File: DSA/Navneetpant_590011569/Day_18/Q1_leetcode.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        ListNode current = head;
        int size = 0 ;
        ArrayList <Integer> al = new ArrayList <> () ;
        while (current!=null){
            size++;
            current=current.next;

        }
        if (size == 0 ){
            return head;
        }
        k = k % size;
        if (k == 0) return head; // no rotation needed

        size = size - k;
        current = head;
        for (int i = 1 ; i < size ; i ++ ) {
            current = current.next;

        }
        ListNode temp = current;
        current=current.next;

        while (current!=null) {
            al.add(current.val);
            current = current.next;
        }

        current = temp;
        current.next=null;
        
        ListNode temper = new ListNode ();
        ListNode head2 = temper;
        
        for (int i = 0 ; i<al.size(); i++ ) {
            temper.next = new ListNode (al.get(i));
            temper = temper.next;

        }
        temper.next = head;
        head = head2;
        return head2.next;
        
    }
}

File: DSA/Prarthana_Baisoya_590024838/Day_18/q1_leetcode.c
struct ListNode* rotateRight(struct ListNode* head, int k)
{
    if(head == NULL || head->next == NULL || k == 0)
        return head;

    int n = 1;
    struct ListNode *temp = head;

    while(temp->next != NULL)
    {
        temp = temp->next;
        n++;
    }

    temp->next = head;

    k = k % n;

    int move = n - k;

    while(move > 0)
    {
        temp = temp->next;
        move--;
    }

    head = temp->next;
    temp->next = NULL;

    return head;
}
